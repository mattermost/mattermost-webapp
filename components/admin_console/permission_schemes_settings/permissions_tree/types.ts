// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type Permission = {
    id: string;
    combined?: boolean;
    permissions: string[];
}
export type Group = {
    id: string;
    permissions: Array<Permission | string>;
}

export type AdditionalValues = {
    edit_post: {
        editTimeLimitButton: JSX.Element;
    };
}
