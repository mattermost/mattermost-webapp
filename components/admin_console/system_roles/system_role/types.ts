// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type ReadAccess = 'read';
export const readAccess: ReadAccess = 'read';
export type WriteAccess = 'write';
export const writeAccess: WriteAccess = 'write';
export type NoAccess = false;
export const noAccess: NoAccess = false;
export type MixedAccess = 'mixed';
export const mixedAccess: MixedAccess = 'mixed';

export type PermissionAccess = ReadAccess | WriteAccess | NoAccess;
export type PermissionsToUpdate = Record<string, ReadAccess | WriteAccess | NoAccess>;
export type PermissionToUpdate = {
    name: string;
    value: PermissionAccess;
};

export type SystemSection = {
    name: string;
    hasDescription?: boolean;
    subsections?: SystemSection[];
    disabled?: boolean;
}
