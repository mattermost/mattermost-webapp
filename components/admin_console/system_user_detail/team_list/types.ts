// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Team, TeamMembership} from '@mattermost/types/teams';

export type TeamWithMembership = Team & TeamMembership;
