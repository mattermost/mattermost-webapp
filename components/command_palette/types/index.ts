// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export enum CommandPaletteEntities {
    Channel = 'channel',
    Playbooks = 'playbooks',
    Boards = 'boards',
    Messages = 'messages',
    Files = 'files',
    Runs = 'runs',
    GoTo = 'goTo',
}

export enum ChannelType {
    DM_CHANNEL= 'D',
    GM_CHANNEL= 'G',
    OPEN_CHANNEL= 'O',
    PRIVATE_CHANNEL= 'P',
}

export enum BoardsType {
    OPEN_CHANNEL= 'O',
    PRIVATE_CHANNEL= 'P',
}

export enum PlaybooksType {
    OPEN_CHANNEL= 'O',
    PRIVATE_CHANNEL= 'P',
}

