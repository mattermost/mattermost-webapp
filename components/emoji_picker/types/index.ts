// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {EmojiCategory, Emoji} from 'mattermost-redux/types/emojis';

export type Category = {
    className: string;
    emojiIds?: string[];
    id: string;
    message: string;
    name: EmojiCategory;
    offset: number;
};

export type Categories = Record<EmojiCategory, Category>;

export type CategoryOrEmojiRow = {
    type: 'categoryHeaderRow' | 'emojisRow';
    row: EmojiCategory | Emoji[];
}
