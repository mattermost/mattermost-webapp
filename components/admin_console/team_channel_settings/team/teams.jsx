// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {createSelector} from 'reselect';

import {getTeams as fetchTeams} from 'mattermost-redux/actions/teams';
import {getTeams} from 'mattermost-redux/selectors/entities/teams';

import {t} from 'utils/i18n';

import TeamList from './team_list.jsx';

const getSortedListOfTeams = createSelector(
    getTeams,
    (teams) => Object.values(teams).sort((a, b) => a.name.localeCompare(b.name))
);

function mapStateToProps(state) {
    return {
        data: getSortedListOfTeams(state),
        total: 100,
        emptyListTextId: t('admin.team_settings.team_list.no_teams_found'),
        emptyListTextDefaultMessage: 'No teams found',
    };
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            getData: fetchTeams,
        }, dispatch),
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(TeamList);
