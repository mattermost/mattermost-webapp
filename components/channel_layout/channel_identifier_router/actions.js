// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {joinChannel, getChannelByNameAndTeamName, markGroupChannelOpen, fetchMyChannelsAndMembers} from 'mattermost-redux/actions/channels';
import {getUser, getUserByUsername, getUserByEmail} from 'mattermost-redux/actions/users';
import {getTeamByName} from 'mattermost-redux/selectors/entities/teams';
import {getCurrentUserId, getUserByUsername as selectUserByUsername, getUser as selectUser, getUserByEmail as selectUserByEmail} from 'mattermost-redux/selectors/entities/users';
import {getChannelByName, getOtherChannels, getChannel, getChannelsNameMapInTeam, getRedirectChannelNameForTeam} from 'mattermost-redux/selectors/entities/channels';

import {Constants} from 'utils/constants';
import {openDirectChannelToUserId} from 'actions/channel_actions';
import * as GlobalActions from 'actions/global_actions.jsx';
import * as Utils from 'utils/utils.jsx';

const LENGTH_OF_ID = 26;
const LENGTH_OF_GROUP_ID = 40;
const LENGTH_OF_USER_ID_PAIR = 54;
const USER_ID_PAIR_REGEXP = new RegExp(`^[a-zA-Z0-9]{${LENGTH_OF_ID}}__[a-zA-Z0-9]{${LENGTH_OF_ID}}$`);

export function onChannelByIdentifierEnter({match, history}) {
    return async (dispatch, getState) => {
        const state = getState();
        const {path, identifier, team} = match.params;

        if (!identifier) {
            return;
        }

        const teamObj = getTeamByName(state, team);
        if (!teamObj) {
            return;
        }

        const channelPath = getPathFromIdentifier(state, path, identifier);

        switch (channelPath) {
        case 'channel_name':
            dispatch(goToChannelByChannelName(match, history));
            break;
        case 'channel_id':
            dispatch(goToChannelByChannelId(match, history));
            break;
        case 'group_channel_group_id':
            dispatch(goToGroupChannelByGroupId(match, history));
            break;
        case 'direct_channel_username':
            dispatch(goToDirectChannelByUsername(match, history));
            break;
        case 'direct_channel_email':
            dispatch(goToDirectChannelByEmail(match, history));
            break;
        case 'direct_channel_user_ids':
            dispatch(goToDirectChannelByUserIds(match, history));
            break;
        case 'direct_channel_user_id':
            dispatch(goToDirectChannelByUserId(match, history, identifier));
            break;
        case 'error':
            await dispatch(fetchMyChannelsAndMembers(teamObj.id));
            handleError(match, history, getRedirectChannelNameForTeam(state, teamObj.id));
            break;
        }
    };
}

export function getPathFromIdentifier(state, path, identifier) {
    if (path === 'channels') {
        // It's hard to tell an ID apart from a channel name of the same length, so check first if
        // the identifier matches a channel that we have
        const channelsByName = getChannelByName(state, identifier);
        const moreChannelsByName = getOtherChannels(state).find((chan) => chan.name === identifier);

        if (identifier.length === LENGTH_OF_ID) {
            return channelsByName || moreChannelsByName ? 'channel_name' : 'channel_id';
        } else if (
            (!channelsByName && !moreChannelsByName && identifier.length === LENGTH_OF_GROUP_ID) ||
            (
                (channelsByName && channelsByName.type === Constants.GM_CHANNEL) ||
                (moreChannelsByName && moreChannelsByName.type === Constants.GM_CHANNEL)
            )
        ) {
            return 'group_channel_group_id';
        } else if (isDirectChannelIdentifier(identifier)) {
            return 'direct_channel_user_ids';
        }
        return 'channel_name';
    } else if (path === 'messages') {
        if (identifier.indexOf('@') === 0) {
            return 'direct_channel_username';
        } else if (identifier.indexOf('@') > 0) {
            return 'direct_channel_email';
        } else if (identifier.length === LENGTH_OF_ID) {
            return 'direct_channel_user_id';
        } else if (identifier.length === LENGTH_OF_GROUP_ID) {
            return 'group_channel_group_id';
        }
        return 'error';
    }

    return 'error';
}

export function goToChannelByChannelId(match, history) {
    return async (dispatch, getState) => {
        const state = getState();
        const {team, identifier} = match.params;
        const channelId = identifier.toLowerCase();

        let channel = getChannel(state, channelId);
        const member = state.entities.channels.myMembers[channelId];
        const teamObj = getTeamByName(state, team);
        if (!channel || !member) {
            const {data, error} = await dispatch(joinChannel(getCurrentUserId(state), teamObj.id, channelId, null));
            if (error) {
                await dispatch(fetchMyChannelsAndMembers(teamObj.id));
                handleChannelJoinError(match, history, getRedirectChannelNameForTeam(state, teamObj.id));
                return;
            }
            channel = data.channel;
        }

        if (channel.type === Constants.DM_CHANNEL) {
            dispatch(goToDirectChannelByUserId(match, history, Utils.getUserIdFromChannelId(channel.name, getCurrentUserId(state))));
        } else if (channel.type === Constants.GM_CHANNEL) {
            history.replace(`/${team}/messages/${channel.name}`);
        } else {
            history.replace(`/${team}/channels/${channel.name}`);
        }
    };
}

export function goToChannelByChannelName(match, history) {
    return async (dispatch, getState) => {
        const state = getState();
        const {team, identifier} = match.params;
        const channelName = identifier.toLowerCase();

        const teamObj = getTeamByName(state, team);
        if (!teamObj) {
            return;
        }

        let channel = getChannelsNameMapInTeam(state, teamObj.id)[channelName];
        let member;
        if (channel) {
            member = state.entities.channels.myMembers[channel.id];
        }

        if (!channel || !member) {
            const {data, error: joinError} = await dispatch(joinChannel(getCurrentUserId(state), teamObj.id, null, channelName));
            if (joinError) {
                const {data: data2, error: getChannelError} = await dispatch(getChannelByNameAndTeamName(team, channelName, true));
                if (getChannelError || data2.delete_at === 0) {
                    await dispatch(fetchMyChannelsAndMembers(teamObj.id));
                    handleChannelJoinError(match, history, getRedirectChannelNameForTeam(state, teamObj.id));
                    return;
                }
                channel = data2;
            } else {
                channel = data.channel;
            }
        }

        if (channel.type === Constants.DM_CHANNEL) {
            dispatch(goToDirectChannelByUserIds(match, history));
        } else if (channel.type === Constants.GM_CHANNEL) {
            history.replace(`/${team}/messages/${channel.name}`);
        } else {
            doChannelChange(channel);
        }
    };
}

function goToDirectChannelByUsername(match, history) {
    return async (dispatch, getState) => {
        const state = getState();
        const {team, identifier} = match.params;
        const username = identifier.slice(1, identifier.length).toLowerCase();
        const teamObj = getTeamByName(state, team);

        let user = selectUserByUsername(state, username);
        if (!user) {
            const {data, error} = await dispatch(getUserByUsername(username));
            if (error) {
                await dispatch(fetchMyChannelsAndMembers(teamObj.id));
                handleError(match, history, getRedirectChannelNameForTeam(state, teamObj.id));
                return;
            }
            user = data;
        }

        const {error, data} = await dispatch(openDirectChannelToUserId(user.id));
        if (error) {
            await dispatch(fetchMyChannelsAndMembers(teamObj.id));
            handleError(match, history, getRedirectChannelNameForTeam(state, teamObj.id));
            return;
        }

        doChannelChange(data);
    };
}

export function goToDirectChannelByUserId(match, history, userId) {
    return async (dispatch, getState) => {
        const state = getState();
        const {team} = match.params;
        const teamObj = getTeamByName(state, team);

        let user = selectUser(state, userId);
        if (!user) {
            const {data, error} = await dispatch(getUser(userId));
            if (error) {
                await dispatch(fetchMyChannelsAndMembers(teamObj.id));
                handleError(match, history, getRedirectChannelNameForTeam(state, teamObj.id));
                return;
            }
            user = data;
        }

        history.replace(`/${team}/messages/@${user.username}`);
    };
}

export function goToDirectChannelByUserIds(match, history) {
    return async (dispatch, getState) => {
        const state = getState();
        const {team, identifier} = match.params;
        const userId = Utils.getUserIdFromChannelId(identifier.toLowerCase(), getCurrentUserId(getState()));
        const teamObj = getTeamByName(state, team);

        let user = selectUser(state, userId);
        if (!user) {
            const {data, error} = await dispatch(getUser(userId));
            if (error) {
                await dispatch(fetchMyChannelsAndMembers(teamObj.id));
                handleError(match, history, getRedirectChannelNameForTeam(state, teamObj.id));
                return;
            }
            user = data;
        }

        history.replace(`/${team}/messages/@${user.username}`);
    };
}

export function goToDirectChannelByEmail(match, history) {
    return async (dispatch, getState) => {
        const state = getState();
        const {team, identifier} = match.params;
        const email = identifier.toLowerCase();
        const teamObj = getTeamByName(state, team);

        let user = selectUserByEmail(state, email);
        if (!user) {
            const {data, error} = await dispatch(getUserByEmail(email));
            if (error) {
                await dispatch(fetchMyChannelsAndMembers(teamObj.id));
                handleError(match, history, getRedirectChannelNameForTeam(state, teamObj.id));
                return;
            }
            user = data;
        }

        history.replace(`/${team}/messages/@${user.username}`);
    };
}

function goToGroupChannelByGroupId(match, history) {
    return async (dispatch, getState) => {
        const state = getState();
        const {identifier, team} = match.params;
        const groupId = identifier.toLowerCase();

        history.replace(match.url.replace('/channels/', '/messages/'));

        let channel = getChannelByName(state, groupId);
        const teamObj = getTeamByName(state, team);
        if (!channel) {
            const {data, error} = await dispatch(joinChannel(getCurrentUserId(state), teamObj.id, null, groupId));
            if (error) {
                await dispatch(fetchMyChannelsAndMembers(teamObj.id));
                handleError(match, history, getRedirectChannelNameForTeam(state, teamObj.id));
                return;
            }
            channel = data.channel;
        }

        dispatch(markGroupChannelOpen(channel.id));

        doChannelChange(channel);
    };
}

function doChannelChange(channel) {
    GlobalActions.emitChannelClickEvent(channel);
}

function handleError(match, history, defaultChannel) {
    const {team} = match.params;
    history.push(team ? `/${team}/channels/${defaultChannel}` : '/');
}

function isDirectChannelIdentifier(identifier) {
    return identifier.length === LENGTH_OF_USER_ID_PAIR && USER_ID_PAIR_REGEXP.test(identifier);
}

async function handleChannelJoinError(match, history, defaultChannel) {
    const {team} = match.params;
    history.push(team ? `/error?type=channel_not_found&returnTo=/${team}/channels/${defaultChannel}` : '/');
}