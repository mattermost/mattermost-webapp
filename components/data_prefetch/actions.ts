// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Dispatch} from 'redux';

import {getChannelIdsForCurrentTeam} from 'mattermost-redux/selectors/entities/channels';

import {trackEvent} from 'actions/telemetry_actions';

import {GlobalState} from 'types/store';

let isFirstPreload = true;

export function trackPreloadedChannels(prefetchQueueObj: Record<string, string[]>) {
    return (dispatch: Dispatch, getState: () => GlobalState) => {
        const state = getState();
        const channelIdsForTeam = getChannelIdsForCurrentTeam(state);

        trackEvent('performance', 'preloaded_channels', {
            numHigh: prefetchQueueObj[1]?.length || 0,
            numMedium: prefetchQueueObj[2]?.length || 0,
            numLow: prefetchQueueObj[3]?.length || 0,

            numTotal: channelIdsForTeam.length,

            // Tracks whether this is the first team that we've preloaded channels for in this session since
            // the first preload will likely include DMs and GMs
            isFirstPreload,
        });

        isFirstPreload = false;
    };
}
