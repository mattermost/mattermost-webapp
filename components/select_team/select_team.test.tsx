// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {Team} from 'mattermost-redux/types/teams';

import SelectTeam, {TEAMS_PER_PAGE} from 'components/select_team/select_team';

import {emitUserLoggedOutEvent} from 'actions/global_actions';

jest.mock('actions/global_actions', () => ({
    emitUserLoggedOutEvent: jest.fn(),
}));

jest.mock('utils/policy_roles_adapter', () => ({
    mappingValueFromRoles: jest.fn(),
}));

describe('components/select_team/SelectTeam', () => {
    const baseProps = {
        currentUserRoles: 'system_admin',
        currentUserId: 'test',
        isMemberOfTeam: true,
        listableTeams: [
            {id: 'team_id_1', delete_at: 0, name: 'team-a', display_name: 'Team A', allow_open_invite: true} as Team,
            {id: 'team_id_2', delete_at: 0, name: 'b-team', display_name: 'B Team', allow_open_invite: true} as Team,
        ],
        siteName: 'Mattermost',
        canCreateTeams: false,
        canManageSystem: true,
        canJoinPublicTeams: true,
        canJoinPrivateTeams: false,
        history: {push: jest.fn()},
        actions: {
            getTeams: jest.fn().mockResolvedValue({}),
            loadRolesIfNeeded: jest.fn(),
            addUserToTeam: jest.fn().mockResolvedValue({data: true}),
        },
        totalTeamsCount: 15,
    };

    test('should match snapshot', () => {
        const props = {...baseProps};
        const wrapper = shallow(<SelectTeam {...props}/>);
        expect(wrapper).toMatchSnapshot();

        // on componentDidMount
        expect(props.actions.getTeams).toHaveBeenCalledTimes(1);
        expect(props.actions.getTeams).toHaveBeenCalledWith(0, TEAMS_PER_PAGE, true);
        expect(props.actions.loadRolesIfNeeded).toHaveBeenCalledWith(baseProps.currentUserRoles.split(' '));
    });

    test('should match snapshot, on loading', () => {
        const wrapper = shallow<SelectTeam>(
            <SelectTeam {...baseProps}/>,
        );
        wrapper.setState({loadingTeamId: 'loading_team_id'});
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, on error', () => {
        const wrapper = shallow<SelectTeam>(
            <SelectTeam {...baseProps}/>,
        );
        wrapper.setState({error: {message: 'error message'}});
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, on no joinable team but can create team', () => {
        const props = {...baseProps, listableTeams: []};
        const wrapper = shallow<SelectTeam>(
            <SelectTeam {...props}/>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, on no joinable team and is not system admin nor can create team', () => {
        const props = {...baseProps, listableTeams: [], currentUserRoles: '', canManageSystem: false, canCreateTeams: false};
        const wrapper = shallow<SelectTeam>(
            <SelectTeam {...props}/>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, on no joinable team and user is guest', () => {
        const props = {...baseProps, listableTeams: [], currentUserRoles: '', currentUserIsGuest: true, canManageSystem: false, canCreateTeams: false};
        const wrapper = shallow<SelectTeam>(
            <SelectTeam {...props}/>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match state and call addUserToTeam on handleTeamClick', async () => {
        const wrapper = shallow<SelectTeam>(
            <SelectTeam {...baseProps}/>,
        );
        await wrapper.instance().handleTeamClick({id: 'team_id'} as any);
        expect(wrapper.state('loadingTeamId')).toEqual('team_id');
        expect(baseProps.actions.addUserToTeam).toHaveBeenCalledTimes(1);
    });

    test('should call emitUserLoggedOutEvent on handleLogoutClick', () => {
        const wrapper = shallow<SelectTeam>(
            <SelectTeam {...baseProps}/>,
        );
        wrapper.instance().handleLogoutClick({preventDefault: jest.fn()} as any);
        expect(emitUserLoggedOutEvent).toHaveBeenCalledTimes(1);
        expect(emitUserLoggedOutEvent).toHaveBeenCalledWith('/login');
    });

    test('should match state on clearError', () => {
        const wrapper = shallow<SelectTeam>(
            <SelectTeam {...baseProps}/>,
        );
        wrapper.setState({error: {message: 'error message'}});
        wrapper.instance().clearError({preventDefault: jest.fn()} as any);
        expect(wrapper.state('error')).toBeNull();
    });
});
