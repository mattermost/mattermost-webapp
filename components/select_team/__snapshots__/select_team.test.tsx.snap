// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/select_team/SelectTeam should match snapshot 1`] = `
<div>
  <Connect(AnnouncementBarController) />
  <BackButton
    url="/"
  />
  <div
    className="col-sm-12"
  >
    <div
      className="select-team__container signup-team__container"
    >
      <img
        alt="signup team logo"
        className="signup-team-logo"
        src={null}
      />
      <SiteNameAndDescription
        siteName="Mattermost"
      />
      <div
        className="signup__content"
        id="teamsYouCanJoinContent"
      >
        <h4>
          <FormattedMessage
            defaultMessage="Teams you can join: "
            id="signup_team.join_open"
          />
        </h4>
        <InfiniteScroll
          bufferValue={280}
          callBack={[Function]}
          endOfDataMessage=""
          itemsPerPage={30}
          loaderStyle={
            Object {
              "height": "40px",
              "padding": "0px",
            }
          }
          pageNumber={0}
          styleClass="signup-team-all"
          totalItems={15}
        >
          <div
            className="signup-team-dir-err"
          >
            <div>
              <FormattedMessage
                defaultMessage="No teams are available to join. Please create a new team or ask your administrator for an invite."
                id="signup_team.no_open_teams_canCreate"
              />
            </div>
          </div>
        </InfiniteScroll>
      </div>
      <Connect(SystemPermissionGate)
        permissions={
          Array [
            "create_team",
          ]
        }
      >
        <div
          className="margin--extra"
          style={
            Object {
              "marginTop": "0.5em",
            }
          }
        >
          <Link
            className="signup-team-login"
            id="createNewTeamLink"
            onClick={[Function]}
            to="/create_team"
          >
            <FormattedMessage
              defaultMessage="Create a team"
              id="login.createTeam"
            />
          </Link>
        </div>
      </Connect(SystemPermissionGate)>
      <Connect(SystemPermissionGate)
        permissions={
          Array [
            "manage_system",
          ]
        }
      >
        <div
          className="mt-8 hidden-xs"
        >
          <Link
            className="signup-team-login"
            onClick={[Function]}
            to="/admin_console"
          >
            <FormattedMessage
              defaultMessage="Go to System Console"
              id="signup_team_system_console"
            />
          </Link>
        </div>
      </Connect(SystemPermissionGate)>
    </div>
  </div>
</div>
`;

exports[`components/select_team/SelectTeam should match snapshot, on error 1`] = `
<div>
  <Connect(AnnouncementBarController) />
  <BackButton
    onClick={[Function]}
    url="/"
  />
  <div
    className="col-sm-12"
  >
    <div
      className="select-team__container signup-team__container"
    >
      <img
        alt="signup team logo"
        className="signup-team-logo"
        src={null}
      />
      <SiteNameAndDescription
        siteName="Mattermost"
      />
      <div
        className="signup__content"
      >
        <div
          className="form-group has-error"
        >
          <label
            className="control-label"
          >
            <Component />
          </label>
        </div>
      </div>
      <Connect(SystemPermissionGate)
        permissions={
          Array [
            "create_team",
          ]
        }
      >
        <div
          className="margin--extra"
          style={
            Object {
              "marginTop": "0.5em",
            }
          }
        >
          <Link
            className="signup-team-login"
            id="createNewTeamLink"
            onClick={[Function]}
            to="/create_team"
          >
            <FormattedMessage
              defaultMessage="Create a team"
              id="login.createTeam"
            />
          </Link>
        </div>
      </Connect(SystemPermissionGate)>
      <Connect(SystemPermissionGate)
        permissions={
          Array [
            "manage_system",
          ]
        }
      >
        <div
          className="mt-8 hidden-xs"
        >
          <Link
            className="signup-team-login"
            onClick={[Function]}
            to="/admin_console"
          >
            <FormattedMessage
              defaultMessage="Go to System Console"
              id="signup_team_system_console"
            />
          </Link>
        </div>
      </Connect(SystemPermissionGate)>
    </div>
  </div>
</div>
`;

exports[`components/select_team/SelectTeam should match snapshot, on loading 1`] = `
<div>
  <Connect(AnnouncementBarController) />
  <BackButton
    url="/"
  />
  <div
    className="col-sm-12"
  >
    <div
      className="select-team__container signup-team__container"
    >
      <img
        alt="signup team logo"
        className="signup-team-logo"
        src={null}
      />
      <SiteNameAndDescription
        siteName="Mattermost"
      />
      <LoadingScreen
        position="relative"
        style={Object {}}
      />
      <Connect(SystemPermissionGate)
        permissions={
          Array [
            "create_team",
          ]
        }
      >
        <div
          className="margin--extra"
          style={
            Object {
              "marginTop": "0.5em",
            }
          }
        >
          <Link
            className="signup-team-login"
            id="createNewTeamLink"
            onClick={[Function]}
            to="/create_team"
          >
            <FormattedMessage
              defaultMessage="Create a team"
              id="login.createTeam"
            />
          </Link>
        </div>
      </Connect(SystemPermissionGate)>
      <Connect(SystemPermissionGate)
        permissions={
          Array [
            "manage_system",
          ]
        }
      >
        <div
          className="mt-8 hidden-xs"
        >
          <Link
            className="signup-team-login"
            onClick={[Function]}
            to="/admin_console"
          >
            <FormattedMessage
              defaultMessage="Go to System Console"
              id="signup_team_system_console"
            />
          </Link>
        </div>
      </Connect(SystemPermissionGate)>
    </div>
  </div>
</div>
`;

exports[`components/select_team/SelectTeam should match snapshot, on no joinable team and is not system admin nor can create team 1`] = `
<div>
  <Connect(AnnouncementBarController) />
  <BackButton
    url="/"
  />
  <div
    className="col-sm-12"
  >
    <div
      className="select-team__container signup-team__container"
    >
      <img
        alt="signup team logo"
        className="signup-team-logo"
        src={null}
      />
      <SiteNameAndDescription
        siteName="Mattermost"
      />
      <div
        className="signup__content"
        id="teamsYouCanJoinContent"
      >
        <h4>
          <FormattedMessage
            defaultMessage="Teams you can join: "
            id="signup_team.join_open"
          />
        </h4>
        <InfiniteScroll
          bufferValue={280}
          callBack={[Function]}
          endOfDataMessage=""
          itemsPerPage={30}
          loaderStyle={
            Object {
              "height": "40px",
              "padding": "0px",
            }
          }
          pageNumber={0}
          styleClass="signup-team-all"
          totalItems={15}
        >
          <div
            className="signup-team-dir-err"
          >
            <div>
              <Connect(SystemPermissionGate)
                permissions={
                  Array [
                    "create_team",
                  ]
                }
              >
                <FormattedMessage
                  defaultMessage="No teams are available to join. Please create a new team or ask your administrator for an invite."
                  id="signup_team.no_open_teams_canCreate"
                />
              </Connect(SystemPermissionGate)>
              <Connect(SystemPermissionGate)
                invert={true}
                permissions={
                  Array [
                    "create_team",
                  ]
                }
              >
                <FormattedMessage
                  defaultMessage="No teams are available to join. Please ask your administrator for an invite."
                  id="signup_team.no_open_teams"
                />
              </Connect(SystemPermissionGate)>
            </div>
          </div>
        </InfiniteScroll>
      </div>
      <Connect(SystemPermissionGate)
        permissions={
          Array [
            "create_team",
          ]
        }
      >
        <div
          className="margin--extra"
          style={
            Object {
              "marginTop": "0.5em",
            }
          }
        >
          <Link
            className="signup-team-login"
            id="createNewTeamLink"
            onClick={[Function]}
            to="/create_team"
          >
            <FormattedMessage
              defaultMessage="Create a team"
              id="login.createTeam"
            />
          </Link>
        </div>
      </Connect(SystemPermissionGate)>
      <Connect(SystemPermissionGate)
        permissions={
          Array [
            "manage_system",
          ]
        }
      >
        <div
          className="mt-8 hidden-xs"
        >
          <Link
            className="signup-team-login"
            onClick={[Function]}
            to="/admin_console"
          >
            <FormattedMessage
              defaultMessage="Go to System Console"
              id="signup_team_system_console"
            />
          </Link>
        </div>
      </Connect(SystemPermissionGate)>
    </div>
  </div>
</div>
`;

exports[`components/select_team/SelectTeam should match snapshot, on no joinable team and user is guest 1`] = `
<div>
  <Connect(AnnouncementBarController) />
  <BackButton
    url="/"
  />
  <div
    className="col-sm-12"
  >
    <div
      className="select-team__container signup-team__container"
    >
      <img
        alt="signup team logo"
        className="signup-team-logo"
        src={null}
      />
      <SiteNameAndDescription
        siteName="Mattermost"
      />
      <div
        className="signup__content"
      >
        <div
          className="form-group has-error"
        >
          <label
            className="control-label"
          >
            <FormattedMessage
              defaultMessage="Your guest account has no channels assigned. Please contact an administrator."
              id="signup_team.guest_without_channels"
            />
          </label>
        </div>
      </div>
      <Connect(SystemPermissionGate)
        permissions={
          Array [
            "create_team",
          ]
        }
      >
        <div
          className="margin--extra"
          style={
            Object {
              "marginTop": "0.5em",
            }
          }
        >
          <Link
            className="signup-team-login"
            id="createNewTeamLink"
            onClick={[Function]}
            to="/create_team"
          >
            <FormattedMessage
              defaultMessage="Create a team"
              id="login.createTeam"
            />
          </Link>
        </div>
      </Connect(SystemPermissionGate)>
      <Connect(SystemPermissionGate)
        permissions={
          Array [
            "manage_system",
          ]
        }
      >
        <div
          className="mt-8 hidden-xs"
        >
          <Link
            className="signup-team-login"
            onClick={[Function]}
            to="/admin_console"
          >
            <FormattedMessage
              defaultMessage="Go to System Console"
              id="signup_team_system_console"
            />
          </Link>
        </div>
      </Connect(SystemPermissionGate)>
    </div>
  </div>
</div>
`;

exports[`components/select_team/SelectTeam should match snapshot, on no joinable team but can create team 1`] = `
<div>
  <Connect(AnnouncementBarController) />
  <BackButton
    url="/"
  />
  <div
    className="col-sm-12"
  >
    <div
      className="select-team__container signup-team__container"
    >
      <img
        alt="signup team logo"
        className="signup-team-logo"
        src={null}
      />
      <SiteNameAndDescription
        siteName="Mattermost"
      />
      <div
        className="signup__content"
        id="teamsYouCanJoinContent"
      >
        <h4>
          <FormattedMessage
            defaultMessage="Teams you can join: "
            id="signup_team.join_open"
          />
        </h4>
        <InfiniteScroll
          bufferValue={280}
          callBack={[Function]}
          endOfDataMessage=""
          itemsPerPage={30}
          loaderStyle={
            Object {
              "height": "40px",
              "padding": "0px",
            }
          }
          pageNumber={0}
          styleClass="signup-team-all"
          totalItems={15}
        >
          <div
            className="signup-team-dir-err"
          >
            <div>
              <FormattedMessage
                defaultMessage="No teams are available to join. Please create a new team or ask your administrator for an invite."
                id="signup_team.no_open_teams_canCreate"
              />
            </div>
          </div>
        </InfiniteScroll>
      </div>
      <Connect(SystemPermissionGate)
        permissions={
          Array [
            "create_team",
          ]
        }
      >
        <div
          className="margin--extra"
          style={
            Object {
              "marginTop": "0.5em",
            }
          }
        >
          <Link
            className="signup-team-login"
            id="createNewTeamLink"
            onClick={[Function]}
            to="/create_team"
          >
            <FormattedMessage
              defaultMessage="Create a team"
              id="login.createTeam"
            />
          </Link>
        </div>
      </Connect(SystemPermissionGate)>
      <Connect(SystemPermissionGate)
        permissions={
          Array [
            "manage_system",
          ]
        }
      >
        <div
          className="mt-8 hidden-xs"
        >
          <Link
            className="signup-team-login"
            onClick={[Function]}
            to="/admin_console"
          >
            <FormattedMessage
              defaultMessage="Go to System Console"
              id="signup_team_system_console"
            />
          </Link>
        </div>
      </Connect(SystemPermissionGate)>
    </div>
  </div>
</div>
`;
